File: e-olymp/DP/O/O.py
#!/usr/bin/env pypy3

import array

MAX_N = 10**5

ans = array.array('i', [MAX_N]*MAX_N)

ans[0] = 0

for i in range(1, MAX_N):
    s = str(i)
    for c in s:
        x = int(c)
        if x > 0 and i - x >= 0:
            ans[i] = min(ans[i], 1+ans[i - x])

print(ans[int(input())])

File: e-olymp/DP/O/O.cpp
#include <bits/stdc++.h>
using namespace std;

constexpr int MAX_N = 1000009;
int ans[MAX_N];

int main() {

  ans[0] = 0;
  for (int i=1;i<MAX_N; i++) {
    ans[i] = MAX_N;
  }

  for (int i=0; i<MAX_N; i++) {
    for (char c : to_string(i)) {
      int x = c - '0';
      if (x > 0 && i - x >= 0) ans[i] = min(ans[i], 1 + ans[i - x]);
    }
  }

  int N;
  cin >> N;
  cout << ans[N] << endl;

  return 0;
}
